File: 0101-0200/0136.Single_Number.js
//16ms



/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function (nums) {
  let arr = nums.sort((a, b) => a - b); // Create a copy of the array and sort it

  for (let i = 0; i <= nums.length - 1; i += 2) { // Iterate by 2
    if (arr[i] !== arr[i + 1]) {
      return arr[i];
    }
  }

  return arr[arr.length - 1];  // Return the first non-matching pair
};




//1ms




/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    return nums.reduce((result, num) => result ^ num, 0);
};




//5ms




/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    const set = new Set();
    const res = [];
    for( let i = 0; i < nums.length; i++ ){
        if(set.has(nums[i])) set.delete(nums[i]);
        else set.add(nums[i]);
    }
    for(let value of set) {
        res.push(value)
    }
    return res[0];
};

File: 0101-0200/0136.Single_Number.java
//1ms



class Solution {
    public int singleNumber(int[] nums) {
        int result=0;
        for(int i=0; i<nums.length; i++) {
            result = result^nums[i];
        }
        return result;
    }
}



//0ms




class Solution {
    static{
        for( int i = 0; i< 500; i++){
            singleNumber(new int[] {});
        }
    }
    public static int singleNumber(int[] nums) {
        int xor = 0;
        for(int num:nums){
            xor^=num;
        }
        return xor;
    }
}
